Subject: [PATCH] SF: Improve phase/vsync offsets on HWC1

On HWC2, SurfaceFlinger races against hardware vsync
on untuned video-mode panels, which can result in a
significant number of dropped frames.
HWC1 does not have this issue, so set the offsets
to 1ms on HWC2 and 0ms for HWC1.

diff --git a/services/surfaceflinger/Android.mk b/services/surfaceflinger/Android.mk
index 170faa8..d40b978 100644
--- a/services/surfaceflinger/Android.mk
+++ b/services/surfaceflinger/Android.mk
@@ -110,16 +110,28 @@
 # [1] https://developer.android.com/studio/profile/systrace.html
 # [2] https://developer.android.com/training/testing/performance.html
 
+# On HWC2, SurfaceFlinger races against hardware vsync
+# on untuned video-mode panels, which can result in a significant number of dropped frames.
+# HWC1 does not have this issue, so set the offsets to 1ms on HWC2 and 0ms for HWC1.
+
 ifneq ($(VSYNC_EVENT_PHASE_OFFSET_NS),)
     LOCAL_CFLAGS += -DVSYNC_EVENT_PHASE_OFFSET_NS=$(VSYNC_EVENT_PHASE_OFFSET_NS)
 else
-    LOCAL_CFLAGS += -DVSYNC_EVENT_PHASE_OFFSET_NS=1000000
+    ifeq ($(TARGET_USES_HWC2),true)
+        LOCAL_CFLAGS += -DVSYNC_EVENT_PHASE_OFFSET_NS=1000000
+    else
+        LOCAL_CFLAGS += -DVSYNC_EVENT_PHASE_OFFSET_NS=0
+    endif
 endif
 
 ifneq ($(SF_VSYNC_EVENT_PHASE_OFFSET_NS),)
     LOCAL_CFLAGS += -DSF_VSYNC_EVENT_PHASE_OFFSET_NS=$(SF_VSYNC_EVENT_PHASE_OFFSET_NS)
 else
-    LOCAL_CFLAGS += -DSF_VSYNC_EVENT_PHASE_OFFSET_NS=1000000
+    ifeq ($(TARGET_USES_HWC2),true)
+        LOCAL_CFLAGS += -DSF_VSYNC_EVENT_PHASE_OFFSET_NS=1000000
+    else
+        LOCAL_CFLAGS += -DSF_VSYNC_EVENT_PHASE_OFFSET_NS=0
+    endif
 endif
 
 ifneq ($(PRESENT_TIME_OFFSET_FROM_VSYNC_NS),)
